_test/test_sqw/test_PixelData.m

```diff
@@ -1113,6 +1114,208 @@ function test_PIXELDATA_raised_if_mem_alloc_argument_is_not_scalar(~)
+        npix_in_page = 11;
+
+        pix = obj.get_pix_with_fake_faccess(data, npix_in_page);
+
+        f = @() pix.move_to_page(ceil(num_pix/npix_in_page) + 1);
+        assertExceptionThrown(f, 'PIXELDATA:move_to_page');
+    end
+
+    function test_move_to_page_throws_if_arg_less_than_1(obj)
+        num_pix = 30;
+        data = rand(PixelData.DEFAULT_NUM_PIX_FIELDS, num_pix);
+        npix_in_page = 11;
+
+        pix = obj.get_pix_with_fake_faccess(data, npix_in_page);
+
+        f = @() pix.move_to_page(0);
+        assertExceptionThrown(f, 'MATLAB:InputParser:ArgumentFailedValidation');
+
+        f = @() pix.move_to_page(-1);
+        assertExceptionThrown(f, 'MATLAB:InputParser:ArgumentFailedValidation');
+    end
+
+    function test_move_to_page_throws_if_arg_is_non_scalar(obj)
+        num_pix = 30;
+        data = rand(PixelData.DEFAULT_NUM_PIX_FIELDS, num_pix);
+        npix_in_page = 11;
+
+        pix = obj.get_pix_with_fake_faccess(data, npix_in_page);
+
+        f = @() pix.move_to_page([1, 2]);
+        assertExceptionThrown(f, 'MATLAB:InputParser:ArgumentFailedValidation');
+    end
+
+    function test_move_to_page_throws_if_arg_is_not_an_int(obj)
+        num_pix = 30;
+        data = rand(PixelData.DEFAULT_NUM_PIX_FIELDS, num_pix);
+        npix_in_page = 11;
+
+        pix = obj.get_pix_with_fake_faccess(data, npix_in_page);
+
+        f = @() pix.move_to_page(1.5);
+        assertExceptionThrown(f, 'MATLAB:InputParser:ArgumentFailedValidation');
+    end
+
+
+    function test_get_pixels_retrieves_data_at_absolute_index(obj)
+        num_pix = 30;
+        data = rand(PixelData.DEFAULT_NUM_PIX_FIELDS, num_pix);
+        npix_in_page = 11;
+
+        start_idx = 9;
+        end_idx = 23;
+
+        pix = obj.get_pix_with_fake_faccess(data, npix_in_page);
+        pix_chunk = pix.get_pixels(start_idx:end_idx);
+
+        assertEqual(pix_chunk.data, data(:, start_idx:end_idx));
+    end
+
+    function test_get_pixels_retrieves_correct_data_at_page_boundary(obj)
+        num_pix = 30;
+        data = rand(PixelData.DEFAULT_NUM_PIX_FIELDS, num_pix);
+        npix_in_page = 10;
+
+        pix = obj.get_pix_with_fake_faccess(data, npix_in_page);
+        pix_chunk1 = pix.get_pixels(1:3);
+        assertEqual(pix_chunk1.data, data(:, 1:3));
+
+        pix_chunk2 = pix.get_pixels(20);
+        assertEqual(pix_chunk2.data, data(:, 20));
+
+        pix_chunk3 = pix.get_pixels(1:1);
+        assertEqual(pix_chunk3.data, data(:, 1));
+    end
+
+    function test_get_pixels_gets_all_data_if_full_range_requested(obj)
+        num_pix = 30;
+        data = rand(PixelData.DEFAULT_NUM_PIX_FIELDS, num_pix);
+        npix_in_page = 11;
+
+        pix = obj.get_pix_with_fake_faccess(data, npix_in_page);
+        pix_chunk = pix.get_pixels(1:num_pix);
+
+        assertEqual(pix_chunk.data, concatenate_pixel_pages(pix));
+    end
+
+    function test_get_pixels_reorders_output_according_to_indices(obj)
```

Comment:
there's a lot of MATLAB magic wrapped up in this test; `randperm`, out-of-order indexing in the `data(:, rand_order)` in the assert.

Is there a way of recasting the test to what is going on is more explicit than:
```
matlab-magic
assertEqual(actual, data(matlab-magic))
```